### DISCUSSION TITLE
Problem compiling Windows version > 2023-06-06 : qt6 (2023-07-15)

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#336] [Q&A]
Since switching to qt6, I can no longer compile the windows version.
What advice can you give me, and what should I check to ensure that the compilation goes smoothly?

at the moment i'm getting a strange error

![pbcompilation](https://github.com/textbrowser/biblioteq/assets/56431039/1bb7dae8-b5dd-4f93-b60d-343222e17b89)

**Comment by textbrowser** [comment]
Will look next time I be on Windows which may be in December or November or earlier or later.

**Comment by textbrowser** [comment]
The strict-overflow is a warning caused by -Wstrict-overflow. Setting is to a weaker value, incrementally decreasing, will resolve the warning. Newer Qt versions and compilers and computers and systems and computers ignite the warnings while previous versions do not. As the compilers become self-aware, they become more agitated with respect to compliance. So, older versions do not surface the warnings while newer versions do. As BQ on Windows intends to focus on Qt 6.5 LTS and newer LTS versions, the warnings must be relaxed.

**Reply by meteos77** [reply]
thanks for the explanation.
It's always interesting to know why a problem occurs.
I got around my problem by recreating a base with version 2023-06-06 and now we can wait for the official version to be released.

**Comment by textbrowser** [comment]
BQ is mature and does not interest me because there are too many requests and needy people and the manner in which I apply it to my life is adequate. And it has to support too many systems and it kind of has too many dependencies like YAZ and PQ and SQLite and MARC and USMARC and Z39.50 and proxies and downloads and it becomes mundane to manage all these systems. And it doesn't leave me with exploration and discovery. It's like reading Earthsea a billion times expecting it to be different but it will not be.

**Reply by meteos77** [reply]
You've done a fantastic job of managing all these features! Thanks again, because thanks to you, we have a great library software that satisfies all the volunteers in my library).

I also understand your point of view, I try to minimize my own requests as much as possible.

**Reply by textbrowser** [reply]
New requests should prioritize creativity. :) If you can solve your problem outside of BiblioteQ, that's great. Creative requests keep the brain active. Repetitive do not.

**Reply by meteos77** [reply]
I try to be creative in my use of BiblioteQ.
After all, my requests to you may not be creative enough for you, but as your knowledge increases ... the harder it is to make creative requests.

**Reply by meteos77** [reply]
After 2 months in operation, we're managing the library well with BiblioteQ.
4380 books; 85 members; 150 loans in progress.

Next step: 600 books out/in in December.
I know that the BA/Discover function is superb, so don't worry.

There's still a lack of time to train volunteers, and volunteers are on call for about 2 hours a month, so there's time to forget in between.

**Comment by textbrowser** [comment]
Well there are annoyances like sed. Modifying a file in-place as it is called requires different syntax on MacOS. That sed is not standard irritates me. Like, a basic tool must be standard. A P2 bit is a P2 bit everywhere in the known Universe.

**Reply by meteos77** [reply]
For me it's really the "quirks" of computing.

i'm trying to use pyinstaller to make an .exe file of my pan2bq tool so that volunteers can transform unimarc files into csv for import into BQ on windows at home.

the syntax of the pyinstaller config file is different on linux and windows ...
/ \ for sub-directories
\ end of lin line or ^ (win)
:(lin) ; (win)

finally, do the job 2 times to manage the 2 systems.

So for those of you who manage MacOS and other systems, I can understand the difficulties of using a program like BiblioteQ.

Imagine: the builder of car A, the accelerator pedal is on the right, the builder B, the pedal is on the left ...
for computing is the symbol for sub-directories that's it (completely stupid).


I hope AI will make it possible to translate programs automatically.
We make a linux program that works, we have a translator like for deepl languages
windows to recipient ... button ... the conversion is done

Translated with www.DeepL.com/Translator (free version)            
:-)

**Comment by textbrowser** [comment]
Reactive software is a large effort. Most software I've encountered is static. It performs a task. Reactive software is software which has awareness. As you interact with it, it performs tasks elsewhere or creates associations elsewhere. It is not static, dynamic. This software requires creativity and engineering and is expensive to create. AI is not necessary for it. It may help, but people can already make this software. And most users are careless and will not notice it, but if they do they will not value it.
